src/Util/MsgCollector.php

```diff
@@ -0,0 +1,306 @@
+<?php
+/**
+ * Collect messages for display at a later point in the process flow.
+ *
+ * If any message with type "error" is passed in, displaying the errors will result in halting the program
+ * with a non-zero exit code.
+ * If only messages with a lower severity are passed in, displaying the errors will be non-blocking
+ * and will not affect the exit code.
+ *
+ * ---------------------------------------------------------------------------------------------
+ * This class is intended for internal use only and is not part of the public API.
+ * This also means that it has no promise of backward compatibility. Use at your own risk.
+ * ---------------------------------------------------------------------------------------------
+ *
+ * @internal
+ *
+ * @author    Juliette Reinders Folmer <phpcs_nospam@adviesenzo.nl>
+ * @copyright 2025 PHPCSStandards and contributors
+ * @license   https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Util;
+
+use InvalidArgumentException;
+use PHP_CodeSniffer\Exceptions\RuntimeException;
+
+final class MsgCollector
```

Comment:
```suggestion
final class MessageCollector
```

Reply:
@fredden I was in two minds about that as it makes the name for use with the "error level" constants longer too. Did you consider that ?

Reply:
Do you mean that `\PHP_CodeSniffer\Util\MsgCollector::ERROR` becomes `\PHP_CodeSniffer\Util\MessageCollector::ERROR`, or something else? I don't see that as a problem. I see the latter as easier to understand.

For me, I first read the class name "MsgCollector" as "a collector for MSG" and wondered why we were collecting monosodium glutamate. I then realised that it's not collecting a food acid, but messages.

I don't see precedent in the codebase for abbreviated words in class names. Perhaps I missed this in my search just now.

Reply:
> Do you mean that `\PHP_CodeSniffer\Util\MsgCollector::ERROR` becomes `\PHP_CodeSniffer\Util\MessageCollector::ERROR`, or something else? I don't see that as a problem. I see the latter as easier to understand.

I did.

> For me, I first read the class name "MsgCollector" as "a collector for MSG" and wondered why we were collecting monosodium glutamate. I then realised that it's not collecting a food acid, but messages.

😂 

> I don't see precedent in the codebase for abbreviated words in class names. Perhaps I missed this in my search just now.

Fair point and better to fix now than having a breaking change later. Done. _... and now I'm off to fix up another ~20 commits which are lined up to build on top of this PR...._